### PR TITLE
LUTECE-2329: Add File download Servlets

### PR DESCRIPTION
**Pull request by glepine1**

### CODE REVIEW

File: src/java/fr/paris/lutece/portal/service/security/RsaService.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+     * @param data
+     * @return decrypted data
+     * @throws GeneralSecurityException
+     */
+    public static final String decryptRsa( String data ) throws GeneralSecurityException
+    {
+        Cipher cipher = Cipher.getInstance( RSA_PADDING );
+        cipher.init( Cipher.DECRYPT_MODE, RSAKeyPairUtil.getInstance( ).getPrivateKey( ) );
+
+        return new String( cipher.doFinal( Base64.getUrlDecoder( ).decode( data.getBytes( StandardCharsets.UTF_8 ) ) ) );
```
Comment (rzara) [inline src/java/fr/paris/lutece/portal/service/security/RsaService.java:81]: I think encoding handling is still wrong. You should use decode(String), since you used endodeToString, and then use the String constructor with UTF-8 charset (since you encoded a UTF-8 representation of the data).
Maybe add tests with unicode characters outside of ISO-8859-1 ?

File: src/java/fr/paris/lutece/util/rsa/RSAKeyPairUtil.java [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+     */
+    public PublicKey getPublicKey( )
+    {
+        return _publicKey;
+    }
+    
+    private void readKeys( ) throws GeneralSecurityException {
+        if ( DatastoreService.existsKey( DATASTORE_PUBLIC_KEY ) && DatastoreService.existsKey( DATASTORE_PRIVATE_KEY ) )
+        {
+            X509EncodedKeySpec keySpecPublic = new X509EncodedKeySpec(Base64.getDecoder().decode(DatastoreService.getDataValue( DATASTORE_PUBLIC_KEY, "" ).getBytes()));
```
Comment (rzara) [inline src/java/fr/paris/lutece/util/rsa/RSAKeyPairUtil.java:91]: getBytes() is dependent on the environnement. You should use a fixed encoding. It seems that since you used encodeToString, the encoding should be ISO-8859-1. Maybe just use decode(String) ?

File: src/java/fr/paris/lutece/util/rsa/RSAKeyPairUtil.java [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+            PKCS8EncodedKeySpec keySpecPrivate = new PKCS8EncodedKeySpec (Base64.getDecoder().decode(DatastoreService.getDataValue( DATASTORE_PRIVATE_KEY, "" ).getBytes()));
```
Comment (rzara) [inline src/java/fr/paris/lutece/util/rsa/RSAKeyPairUtil.java:92]: same story about encoding

File: src/java/fr/paris/lutece/util/rsa/RSAKeyPairUtil.java [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+            
+            KeyFactory keyFactory = KeyFactory.getInstance("RSA");
+            
+            this._publicKey = keyFactory.generatePublic( keySpecPublic );
+            this._privateKey = keyFactory.generatePrivate( keySpecPrivate );
+        }
+        else
+        {
+            KeyPairGenerator keyGen = KeyPairGenerator.getInstance( "RSA" );
+            keyGen.initialize( 1024 );
```
Comment (rzara) [inline src/java/fr/paris/lutece/util/rsa/RSAKeyPairUtil.java:102]: maybe you should comment on why this size is enough when ANSSI recommends 2048 at least ?

**Review by rzara — COMMENTED** [review]
only reviewed the first commit. It seems most of my comments on the previous PR still stand
